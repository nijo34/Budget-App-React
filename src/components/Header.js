import React from 'react'; 
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
    <header>
        <h1>Budget App</h1>
        <NavLink to ="/" activeClassName="is-active" exact = {true}>HomePage</NavLink>
        <NavLink to ="/create" activeClassName="is-active">Create</NavLink>        
        <button onClick = {startLogout} >Logout</button>
    </header>
)

const mapDispatchToProps = (dispatch) =>({
    startLogout : ()=>dispatch(startLogout())
})

export default connect(undefined , mapDispatchToProps)(Header);