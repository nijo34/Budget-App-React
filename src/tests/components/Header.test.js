import ReactShallowRenderer from "react-test-renderer/shallow";
import React from 'react'
// import toJSON from 'enzyme-to-json'                                 //if not used, then we get a lot of junk data from enzyme within the snapshot which if altered, can lead to faulty test cases.
import { shallow } from 'enzyme'
import Header from '../../components/Header'


test('should render header correctly',()=>{
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()                               //removed enzyme-to-json since it is included in the config file.

    // expect(wrapper.find('h1').text()).toBe('Budget App')
    //  const renderer = new ReactShallowRenderer();              //created a new instance of shallow renderer
    //  renderer.render(<Header />)                               //shallow rendered the Header component
    //  expect(renderer.getRenderOutput()).toMatchSnapshot()      //method used to get the rendered output via the shallow renderer.
                                                               //it always passes the first time, since there is no existing snapshot.
})