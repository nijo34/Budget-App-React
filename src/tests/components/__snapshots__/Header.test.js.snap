// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Budget App
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    HomePage
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create
  </NavLink>
</header>
`;
